src/stream/src/executor/aggregation/agg_impl/foldable.rs

```diff
@@ -127,6 +127,41 @@ where
     }
 }
 
+/// `I64Sum0` sums two primitives by `accumulate` and `retract` functions.
+/// It produces the same type of output as input `S`.
+#[derive(Debug)]
+pub struct I64Sum0 {}
```

Comment:
How about storing a `PrimitiveSummable<i64, i64>` and forwarding the `accumulate` and `retract` to it?

Reply:
Sounds good.